### ISSUE TITLE
add a "tour of the examples" page

### ISSUE THREAD

**Issue by benswift** [extemporelang/extemporelang.github.io#9]
There's some good example code in the `examples` folder of the main Extempore repo, but it's hard to know it's there if it's not pointed out. We should do either (or both) of

- have a "guide to the examples" page in the docs, pointing out the roles of the various examples and where to go to learn how to do specific things
- add a "relevant example code" section to each docs page & guide

In general, we should make sure the examples are up-to-date and display best practice, modern xtlang wherever possible.

**Comment by hellerve** [comment]
This would be amazing! I had some trouble getting a lot of the examples to run properly. I think this is mainly because I’m new to the language, but it was definitely frustrating in the beginning.

**Comment by benswift** [comment]
Yeah, there's an ongoing issue with subtle differences between HEAD & the last release, and therefore the docs not working for both cases. Examples are part of this too (and you touched on this in #13).

Re: the docs tour, if you're on HEAD, which examples did you find most illuminating? Because it's always harder for @digego and I to know where the pain points are and which examples produce the 'aha' moments.

**Comment by hellerve** [comment]
Alright, that makes sense.

Honestly, I eventually gave up on the examples. One of the examples that worked and was immediately gratifying was `examples/core/audio_101.xtm`. I still can’t get the OpenGL things to run.

Now I’m trying to work my way through the website and play around with the environment a little bit! Will keep you posted!

**Comment by digego** [comment]
Did you download the assets?  Most of the OpenGL examples expect to find a top level assets folder.

On 19 July 2018 6:47:29 pm AEST, Veit Heller <notifications@github.com> wrote:
>Alright, that makes sense.
>
>Honestly, I eventually gave up on the examples. One of the examples
>that worked and was immediately gratifying was
>`examples/core/audio_101.xtm`. I still can’t get the OpenGL things to
>run.
>
>Now I’m trying to work my way through the website and play around with
>the environment a little bit! Will keep you posted!
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/extemporelang/extemporelang.github.io/issues/9#issuecomment-406202966

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by hellerve** [comment]
I don’t think so. The example where I first noticed the problem is a `contrib` example—specifically `game-of-life`, so maybe you don’t support that? It seems self-contained enough (the shaders are in `contrib` as well), and I can see a process is created, but it doesn’t show anything or respond to my incantations.

I don’t want to pollute this thread too much with my newbie struggles, though. Is there a chat where I can better pose these questions? Lurk seems somewhat abandoned (and I guess the Europe/Australia timezone diff doesn’t help).

**Comment by digego** [comment]
I'm often on lurk during the Aus day.  Say hi :)
Grab the assets and try xtmrender1-6 examples.  These are the best place to start for 3d graphics.


On 19 July 2018 8:16:13 pm AEST, Veit Heller <notifications@github.com> wrote:
>I don’t think so. The example where I first noticed the problem is a
>`contrib` example—specifically `game-of-life`, so maybe you don’t
>support that? It seems self-contained enough (the shaders are in
>`contrib` as well), and I can see a process is created, but it doesn’t
>show anything or respond to my incantations.
>
>I don’t want to pollute this thread too much with my newbie struggles,
>though. Is there a chat where I can better pose these questions? Lurk
>seems somewhat abandoned (and I guess the Europe/Australia timezone
>diff doesn’t help).
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/extemporelang/extemporelang.github.io/issues/9#issuecomment-406227803

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
